src/CastleAdventure.java

```diff
@@ -3,29 +3,170 @@
 public class CastleAdventure {
 
     public static void main(String[] args) {
-        Scanner scanner = new Scanner(System.in);
-
-        // Нижче перша частина завдання для прикладу
-        // Там де в завданні вказані початкові дані у вигляді масиву - ініціалізуйте окремий масив
-        // Нижче обробляйте масив, додайте логіку обробки
-        // Частина 1: Увійди в Замок
-        int[] secretCode = {7, 14, 28, 56, 112};
-        System.out.println("Введіть 5 чисел, щоб відкрити двері замку:");
-        boolean isCodeCorrect = true;
-
-        for (int i = 0; i < secretCode.length; i++) {
-            int userInput = scanner.nextInt(); // питаємо в користувача чергове число
-            if (userInput != secretCode[i]) { // перевіряємо число проти i-го елемента масиву з правильним кодом
-                isCodeCorrect = false; // якщо натрапляємо на неправильно введене число - усю комбінацію вважаємо невірною
+        System.out.println("Hello brave hero!");
+        System.out.println("Welcome into the 'Castle Adventure' journey!");
+        System.out.println("Your first task is to enter the castle. To open the door you need to enter correct password!");
+        System.out.println("Do you need a hint for password answer? Type 1 for 'yes' or 2 for 'no'");
+        Scanner denScanner = new Scanner(System.in);
+
+        
+        int choice1 = denScanner.nextInt();
+        if (choice1 == 1) {
+            System.out.println("1: How many oceans do we have?");
```

Comment:
That's beautiful, you're the only one who made hints, otherwise I'll keep guessing forever (if I don't have access to the code) 🥇